### PR TITLE
td-paging: Use checked_sub() to avoid overflow in page_table.rs

### PR DESCRIPTION
**Pull request by liuw1**
Fix: #383
Signed-off-by: Wei Liu <wei3.liu@intel.com>

### CODE REVIEW

File: td-paging/src/page_table.rs [hunk 1]
```diff
@@ -108,7 +108,7 @@ pub fn create_mapping_with_flags(
             Size4KiB::SIZE
         };
 
-        sz -= mapped_size;
+        sz = sz.checked_sub(mapped_size).unwrap();
```
Comment (jodh-intel) [inline td-paging/src/page_table.rs:111]: This function returns a `Result<()>` so couldn't this be:

```suggestion
        sz = sz.checked_sub(mapped_size)?;
```
```suggestion
        sz = sz.checked_sub(mapped_size).unwrap();
```

Comment (liuw1) [reply td-paging/src/page_table.rs:111]: 111 | |         sz = sz.checked_sub(mapped_size)?;
    | |                                         ^ cannot use the `?` operator in a function that returns `core::result::Result<(), Error>`

Comment (gaojiaqi7) [reply td-paging/src/page_table.rs:111]: hi @liuw1 , should we return error or unwrap here? Why not change `sz` as a signed integer. If we don't accept `sz` less than `0x1000`, we should check it at https://github.com/confidential-containers/td-shim/blob/main/td-paging/src/page_table.rs#L84

Comment (liuw1) [reply td-paging/src/page_table.rs]: Resolved

**Pull request merged**